server-state-machine.go

```diff
@@ -218,7 +219,8 @@ func (state ServerStateStart) Next(hm *HandshakeMessage) (HandshakeState, []Hand
 
 		nextState := ServerStateStart{
 			Caps:              state.Caps,
-			cookie:            cookie.Cookie,
+			conn:              state.conn,
+			cookieSent:        true,
```

Comment:
Why did you move cookie into conn rather than leaving it in ServerStateStrart?

Reply:
The `conn` is needed as a parameter for `CookieHandler.Generate` and `Validate` anyway.